File: HW3/smallest_largest.cpp
// Copyright 2019 Haoyu Xu xhy@bu.edu
double largest_double() {
	double res;
	long int *pointer = (long int*) &res;
	*pointer = 0x7fefffffffffffff;

	return res;
}

double smallest_double() {
	double res;
	long int *pointer = (long int*) &res;
	*pointer = 0x1;

	return res;
}
  
float largest_single() {
	float res;
	int *pointer = (int*) &res;
	*pointer = 0x7f7fffff;

	return res;
}

float smallest_single() {
	float res;
	int *pointer = (int*) &res;
	*pointer = 0x00000001;

	return res;
}

File: HW3/smallest_largest.py
#!/usr/bin/python
# Copyright 2019 Haoyu Xu xhy@bu.edu
def largest_double():
	a = 2.0**1023
	b = 2 - 2**(-52)
	return a * b

def smallest_double():
	return 2.0**(-1022) * 2**(-52)
  
def largest_single():
	a = 2.0**127
	b = 2 - 2**(-23)
	return a * b

def smallest_single():
	return 2.0**(-126) * 2**(-23)


def main():
	print(largest_double())
	print(smallest_double())
	print(largest_single())
	print(smallest_single())

if __name__ == '__main__':
	main()